filament/include/filament/Engine.h

```diff
@@ -151,6 +167,70 @@ class UTILS_PUBLIC Engine {
 public:
     using Platform = backend::Platform;
     using Backend = backend::Backend;
+    using DriverConfig = backend::Platform::DriverConfig;
+
+    /**
+    * ConfigParams are used to define the memory footprint used by the engine, such as the
+    * command buffer size. ConfigParams can be used to customize engine requirements based 
+    * on the applications needs.
+    */
+    class ConfigParams {
```

Comment:
Make this a simple `struct` with public fields. Also rename to `Config`. e.g.:

```
struct Config {
    uint8_t commandBufferSize = FILAMENT_COMMAND_BUFFER_SIZE_IN_MB;   //!< size of command buffer in MiB
    ...
};
```

Validate the parameters in `Engine::create`. No need to store `DriverConfig` in this structure -- just create one when needed to pass to the driver.

We basically don't want any private fields visible in public headers.